jujugui/static/gui/src/app/components/user-profile/kpi/kpi.js

```diff
@@ -0,0 +1,88 @@
+/*
+This file is part of the Juju GUI, which lets users view and manage Juju
+environments within a graphical interface (https://launchpad.net/juju-gui).
+Copyright (C) 2017 Canonical Ltd.
+
+This program is free software: you can redistribute it and/or modify it under
+the terms of the GNU Affero General Public License version 3, as published by
+the Free Software Foundation.
+
+This program is distributed in the hope that it will be useful, but WITHOUT
+ANY WARRANTY; without even the implied warranties of MERCHANTABILITY,
+SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero
+General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License along
+with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+'use strict';
+
+YUI.add('user-profile-entity-kpi', function() {
+
+  juju.components.UserProfileEntityKPI = React.createClass({
+    displayName: 'Profile KPIs',
+
+    propTypes: {
+      d3: React.PropTypes.object.isRequired,
+      metricTypes: React.PropTypes.array.isRequired,
+      metrics: React.PropTypes.array.isRequired
+    },
+
+    getInitialState: function() {
+      return {
+        currentMetric: []
+      };
+    },
+
+    /**
+      Tell the metric component to render a graph of a specific metric.
+
+      @method _showMetric
+      @param {String} metric The metric to show (selected from metricTypes)
+    */
+    _showMetric: function(metric) {
+      // TODO:
+      // the ability to select a metric does not work properly, and we don't
+      // know enough about the eventual metrics of charms to guess how; for
+      // now, just render the first metric at hand. See also the comment in
+      // in componentDidMount
+      // Makyo 2017-04-03
+      metric = this.props.metrics[0].metric;
+      this.setState({
+        currentMetric: this.props.metrics.filter(d => d.metric === metric)
+      });
```

Comment:
Setting state here seems a bit overkill as it's only used to render the filtered props.metrics.
We could just return the filter and call `this._showMetric` from `render` removing both `getInitialState` and `componentDidMount` with the bonus of not holding state in the component.